src/app/reporting/Engine.cpp

```diff
@@ -489,10 +488,35 @@ void Engine::Run()
 
     if (allReadClean)
     {
-        InteractionModelEngine::GetInstance()->ReleaseClusterInfoList(mpGlobalDirtySet);
+        mGlobalDirtySet.ReleaseAll();
     }
 }
 
+bool Engine::MergeOverlappedAttributePath(ClusterInfo & aAttributePath)
+{
+    bool overlapped = false;
+    mGlobalDirtySet.ForEachActiveObject([&](auto * path) {
+        // If overlapped, we would skip this target path,
+        // --If targetPath is part of previous path, return true
+        // --If previous path is part of target path, update filedid and listindex and mflags with target path, return true
```

Comment:
I suspect this is aiming for something more like this:
```suggestion
        // If we already have a path that is a superset of the provided path.
        //
        // If the provided path is a superset of our of our existing paths, update that existing path to match the
        // provided path.
        //
        // Return whether one of our paths is now a superset of the provided path.
```
and that should probably be API documentation for the function, not documentation here.

Reply:
updated, and relocatd, thanks